flink-filesystems/flink-gs-fs-hadoop/pom.xml

```diff
@@ -214,6 +214,8 @@ under the License.
 										<exclude>META-INF/*.SF</exclude>
 										<exclude>META-INF/*.DSA</exclude>
 										<exclude>META-INF/*.RSA</exclude>
+										<!-- exclude javax.annotation because of GPLv2 license, see https://github.com/apache/flink/pull/15599#issuecomment-850241316 -->
+										<exclude>javax/annotation/**</exclude>
```

Comment:
Instead of filtering it out when the shading happens, can this dependency be excluded when depending on the dependency which pulls it it (e.g. in the `<dependencies>` section)?

Reply:
Yes, the dependency is already excluded in that manner, [here](https://github.com/apache/flink/blob/4050cf320fc6c3ad77ad406689dd8a99af75d594/flink-filesystems/flink-gs-fs-hadoop/pom.xml#L89). And for Java 8 builds, that is sufficient -- there is no licensing problem.

In Java 11 builds, those two `package.html` files still seem to be present and causing problems, somehow. Adding the filter is an attempt to remove them from the jar file in those builds. In a local build of the Java 11 jar file, with this configuration, I can confirm by inspection that no file from `javax.annotation` -- including the `package.html` files -- is present. I don't know how to run the license check locally, though, so I can't test whether this fixes the Java 11 issue or not, without having someone merge the code.

I'm certainly open to alternate suggestions!

Reply:
Hmm, what I would like to understand is why javax.annotation ends up in the final jar if it is already excluded?

Reply:
Agreed, that's the puzzling part.

It doesn't seem that there are any actual code files included in the Java 11 jar, it's just these two `package.html` files. Maybe those files are treated differently in the bundling process, somehow?

Reply:
I think the dependency is pulled in by the `gcs-connector` dependency when using Java 11. Hence, adding the exclusion to this dependency should solve the problem.

Reply:
I've created a quick PR with the changes #18484. PTAL.

Reply:
I had actually just pushed a commit to this PR with those changes, as well: https://github.com/apache/flink/pull/18452/commits/d9507c8a77a6674f949c2f19e0f75c2419b87396.